tests/integration/test_rest.py

```diff
@@ -227,9 +227,9 @@ async def test_internal(rest_async: KafkaRest | None, admin_client: KafkaAdminCl
     assert len(results) == 1
     for result in results:
         assert "error" in result, "Invalid result missing 'error' key"
-        assert result["error"] == "Unrecognized partition"
+        assert result["error"] == "This request is for a topic or partition that does not exist on this broker."
         assert "error_code" in result, "Invalid result missing 'error_code' key"
-        assert result["error_code"] == 1
```

Comment:
`aiokafka` raises an `AssertionError` in this case, which is considered non-retriable, however from `confluent-kafka` an `UnknownTopicOrPartitionError` equivalent is raised, which is considered retriable - hence the difference in the error codes.

I'm not entirely sure which behaviour we'd like to go forward with, the "correct" one with code `2`, or preserve the previous. 🤔